### PR TITLE
Update deployment documentation (closes #1902)

### LINKED ISSUE #1902: Deployment documentation updates
The _Configure GCP_ section specifies that `./scripts/populate_lambda_ssm_parameters.py` should be executed, but its invocation will fail unless:
  * An ElasticSearch instance, `dss-index-dev`, exists.
  * GCP service(/oauth?) keys are created and provided to AWS via `cat credentials.json | ./scripts/set_secret.py --secret-name $GOOGLE_APPLICATION_CREDENTIALS_SECRETS_NAME`

Deployment documentation should be updated to reflect this process and also to note that `AWS_DEFAULT_PROFILE` or related variables should be set if using assumed roles

### PR DESCRIPTION
**Pull request by natanlao**
* Formatting, spelling fixes
* Added note on Terraform
* Clarified GCP authentication flow

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,10 +69,30 @@ AWS command line utility and configure your AWS access credentials.
    that `AWS_DEFAULT_REGION` points to your prefered region.
    These buckets will be created with Terraform, and should not exist before deploying for the first time.
 
+1. Create an AWS ElasticSearch Service instance with the name specified by `$DSS_ES_DOMAIN` (e.g. via `aws es create-elasticsearch-domain --domain-name $DSS_ES_DOMAIN`).
```
Comment (amarjandu) [inline README.md]: we should be utilizing terraform for this to maintain a state of the ES instance, see `$DSS_HOME/infra/elasticsearch`

File: README.md [hunk 2]
```diff
@@ -69,10 +69,30 @@ AWS command line utility and configure your AWS access credentials.
...
+
 #### Configure GCP
 
 1.  Follow the instructions in https://cloud.google.com/sdk/downloads to get the `gcloud` command line utility.
 
+1.  Generate service account keys for setup:
+
+    1) Go to https://console.developers.google.com/apis/credentials (you may have to select Organization and Project
```
Comment (amarjandu) [inline README.md]: use hyperlinks for click flow through document.

File: README.md [hunk 3]
```diff
@@ -69,10 +69,30 @@ AWS command line utility and configure your AWS access credentials.
...
+    again)
+    
+	1) Click *Create Credentials* and select *Service account key*
+    
+	1) For *Key type* choose JSON. 
+    
+
+	1) Place the downloaded JSON file into the project root as `gcp-credentials.json` (or whatever `GOOGLE_APPLICATION_CREDENTIALS_SECRETS_NAME` is set to in your `environment.local`).
```
Comment (amarjandu) [inline README.md]: Try to use a more formal language for a dev doc (or whatever ->  to what the); with that being said i think telling the user that placing it in the root for the project is suffice, in the next code command below this, the user is told to `cat $DSS_HOME/gcp-credendials.json` but if they move the file around following the statement above, that wont work correctly.

File: README.md [hunk 4]
```diff
@@ -69,10 +69,30 @@ AWS command line utility and configure your AWS access credentials.
    that `AWS_DEFAULT_REGION` points to your prefered region.
    These buckets will be created with Terraform, and should not exist before deploying for the first time.
 
+1. Create an AWS ElasticSearch Service instance with the name specified by `$DSS_ES_DOMAIN` (e.g. via `aws es create-elasticsearch-domain --domain-name $DSS_ES_DOMAIN`).
+
 #### Configure GCP
 
 1.  Follow the instructions in https://cloud.google.com/sdk/downloads to get the `gcloud` command line utility.
 
+1.  Generate service account keys for setup:
```
Comment (DailyDreaming) [inline README.md]: Step 1 ought to be creating a service account first.

File: README.md (hunk 1 above)
Comment (natanlao) [reply README.md]: #1960

**Comment by amarjandu** [comment]
Natan its hard for me to see context of what has been changed if commits are squashed before the final merge, please wait till the PR has been approved, it just makes it easier to track changes as the PR gets feedback, otherwise it looks good to me. 
```
                 _
                                    (_)
              |    .
          .   |L  /|   .          _
      _ . |\ _| \--+._/| .       (_)
     / ||\| Y J  )   / |/| ./
    J  |)'( |        ` F`.'/        _
  -<|  F         __     .-<        (_)
    | /       .-'. `.  /-. L___       
    J \      <    \  | | O\|.-'  _   
  _J \  .-    \/ O | | \  |F    (_) 
 '-F  -<_.     \   .-'  `-' L__    
__J  _   _.     >-'  )._.   |-'  
`-|.'   /_.           \_|   F    
  /.-   .                _.<     
 /'    /.'             .'  `\    
  /L  /'   |/      _.-'-\
 /'J       ___.---'\|
   |\  .--' V  | `. `
   |/`. `-.     `._)
      / .-.\
      \ (  `\```

**Comment by natanlao** [comment]
https://allspark.dev.data.humancellatlas.org/HumanCellAtlas/data-store/pipelines/8029

**Pull request merged**